Bug/eosu 734 sandbox json update and save fix

JessTello opened the pull request:
This pull request addresses multiple issues and improvements related to configuration handling and editor behavior.

**Changes included:**

- Fix: Removed validation that prevented proper JSON updates when sandboxes were deleted.
- Fix: Corrected save behavior so both product and platform configs are saved on the first click.
- Fix: Removed redundant Write() call that was executed constantly while the settings window was open.
- Style: Normalized JSON line endings (CRLF → LF) across Android, iOS, and Linux configuration files for consistency.
- Refactor: Removed unnecessary AssetDatabase.Refresh() and SaveAssets() calls to prevent window initialization errors.

JessTello committed:
Refactor: Remove UnityEditor asset refresh to prevent window initialization error

- Deleted calls to AssetDatabase.SaveAssets() and Refresh()
- Added comment to clarify no action is taken

JessTello committed:
Normalize JSON line endings and EOF marker

Fix across Android, Linux and iOS json

JessTello committed:
Fix: Correct save behavior and remove redundant Write call

- Fixed issue where first 'Save' click only stored product config and second click stored platform config
- Removed unnecessary Write() invocation that was repeatedly executed while the window remained open

JessTello committed:
Fix: Ensure JSON updates on sandbox deletion

- Deleted check that prevented proper update after sandbox removal
- Ensured JSON reflects changes immediately when sandboxes are deleted

matt-clarke commented on Assets/Plugins/Source/Editor/EditorWindows/EOSSettingsWindow.cs:
```diff
@@ -255,7 +255,9 @@ private void Save()
-            // reload the product config editor
+            //Update deployment in current platform
+            ProductConfig.Get<ProductConfig>().Environments.TryGetFirstDefinedNamedDeployment(out var namedDep);
+            _platformConfigEditors[_selectedTab].SetDeployment(namedDep.Value);
```
The `TryGet` should return a bool, which needs checking before you try access the out param

JessTello replied:
This has been implemented in the following commit.

matt-clarke commented on Assets/Plugins/Source/Editor/Utility/GUIEditorUtility.cs:
```diff
@@ -993,6 +993,7 @@ private static void RenderSandboxInputs(ref ProductionEnvironments value)
                     if (!productionEnvironmentsCopy.Sandboxes.Remove(item))
                     {
                         // TODO: Tell user why the sandbox could not be removed.
+                        Debug.LogError($"{nameof(GUIEditorUtility)} {nameof(RenderSandboxInputs)}: Could not remove sandbox value .");
```
Updating the message should cover this TODO

**(untested code)**
```suggestion
                        Debug.LogError($"{nameof(GUIEditorUtility)} {nameof(RenderSandboxInputs)}: Failed to find sandbox with name {item.Name} when trying to remove it.");
```

JessTello replied:
This has been implemented in the following commit.

matt-clarke commented on com.playeveryware.eos/Runtime/Core/Config/Config.cs:
```diff
@@ -188,7 +188,7 @@ private void MigrateConfigIfNeeded()
 #if UNITY_EDITOR
-            Write();
+            //Write();
 #endif
```
`#if` block can be deleted

JessTello replied:
This has been implemented in the following commit.

matt-clarke commented on com.playeveryware.eos/Runtime/Core/Config/Config.cs:
```diff
@@ -506,11 +506,6 @@ public virtual async Task WriteAsync(bool prettyPrint = true)
-            // If the json hasn't changed since it was last read, then
-            // take no action.
-            if (json == _lastReadJsonString)
-                return;
```
What item in the description is related to this? Is it this one?
> Fix: Corrected save behavior so both product and platform configs are saved on the first click.

Regardless this is more of a case of not fixing the underlying problem and just removing this check. The real issue is why does this evaluate to true and result in aborting the save

JessTello replied:
That condition was actually preventing the sandboxes from being deleted. I tried setting it to null, but it still didn’t work. Let me check again today with more time, because that condition definitely causes the changes not to be saved when they are deleted, even if  _lastReadJsonString it’s null.

JessTello replied:
You’re absolutely right. Looking at it now with a different perspective than a few days ago, the only thing missing was updating the _lastReadJsonString variable when writing. Thanks for pointing that out! 
This has been implemented in the following commit.

JessTello committed:
Fix: Adjust config write logic and improve sandbox error handling

- Removed unnecessary UNITY_EDITOR condition in Config.
- Updated WriteAsync and Write to ensure proper JSON persistence.
- Improved error messages when sandbox removal fails.
- Enhanced deployment handling for missing named deployments.

billfr-ApexSystems reviewed (approved):
Approved with changes requested
